api/src/config.rs

```diff
@@ -252,6 +269,17 @@ impl BootloaderConfig {
             (frame_buffer, s)
         };
 
+        let (&log_level, s) = split_array_ref(s);
+        let log_level = match u8::from_le_bytes(log_level) {
+            0 => log::LevelFilter::Trace,
+            1 => log::LevelFilter::Debug,
+            2 => log::LevelFilter::Info,
+            3 => log::LevelFilter::Warn,
+            4 => log::LevelFilter::Error,
+            5 => log::LevelFilter::Off,
+            _ => return Err("log_level invalid"),
+        };
```

Comment:
When we create our own `LevelFilter` enum, we should include a `from_u8(value: u8) -> Option<LevelFilter>` method and use it here (so that we have the integer representation in a single place).

Reply:
Thanks for the feedback! 
One question: 
So the kernel must use our implementation of the LevelFilter instead the log's one.
And also we have to make a casting in the uefi and bios crate to convert our LevelFilter into the log's LevelFilter.
Is that what you are saying?

Reply:
The method from_usize() in log crate is a private function. So we need another function (maybe in the common crate) that converts our LevelFilter to u8 , the u8 to usize and a match statement for converting the usize into the log's LevelFitler. And then passing the result to set_max_level().

Reply:
Yeah, what I meant is that we should duplicate the `LevelFilter` enum of the log crate, together with its `from_usize` method. We can then change `usize` to `u8` to avoid some casts.

> So we need another function (maybe in the common crate) that converts our LevelFilter to [...] log's LevelFitler.

Yeah, we need to do this conversion. The simplest way to implement it is probably something like the following:


```rust
fn convert_level(level: LevelFilter) -> log::LevelFilter {
    match level {
        LevelFilter::Error => log::LevelFilter::Error,
        LevelFilter::Warn => log::LevelFilter::Warn,
        LevelFilter::Info => log::LevelFilter::Info,
        [...]
    }
}